src/AspNetCoreModuleV2/InProcessRequestHandler/inprocesshandler.cpp

```diff
@@ -92,16 +94,33 @@ REQUEST_NOTIFICATION_STATUS IN_PROCESS_HANDLER::ServerShutdownMessage()
 VOID
 IN_PROCESS_HANDLER::NotifyDisconnect()
 {
+    m_moduleTracer.RequestDisconnect();
+
     if (m_pApplication->QueryBlockCallbacksIntoManaged() ||
         m_fManagedRequestComplete)
     {
         return;
     }
 
-    m_moduleTracer.RequestDisconnect();
+    // For client disconnects, we are allowed to call m_pDisconnectHandler() multiple times,
```

Comment:
Are we ever going to call m_pDisconnectHandler() multiple times with the current logic we have?